common/beerocks/hostapd/configuration.cpp

```diff
@@ -198,6 +197,43 @@ void Configuration::disable_all_ap_vaps() {
 
 const std::string &Configuration::get_last_message() const { return m_last_message; }
 
+std::tuple<bool, std::vector<std::string> *>
+Configuration::get_vap(const std::string &calling_function, const std::string &vap)
+{
+    // search for the requested vap
+    auto existing_vap = m_hostapd_config_vaps.find(vap);
+
+    if (existing_vap == m_hostapd_config_vaps.end()) {
+        m_last_message = calling_function + " couldn't find requested vap: " + vap;
+        m_ok           = false;
+        return std::make_tuple(false, nullptr);
+    }
+
+    m_ok = true;
+    return std::make_tuple(true, &existing_vap->second);
+}
+
+bool Configuration::is_key_in_line(const std::string &line, const std::string &key) const
+{
+    // we need to make sure when searching for example
+    // for "ssid", to ignore cases like:
+    // multi_ap_backhaul_ssid="Multi-AP-24G-2"
+    //                   ^^^^^
+    // bssid=02:9A:96:FB:59:11
+    //  ^^^^^
+    // and we need to take into consideration
+    // that the key might be or might not be commented.
+    // so the search algorithm is:
+    // - find the requested key and
+    // - make sure it is either on the first position
+    // or it has a comment sign just before it
+    auto found_pos = line.rfind(key);
+    bool ret = found_pos != std::string::npos && (found_pos == 0 || line.at(found_pos - 1) == '#');
```

Comment:
I think it would be better to remove all the comments when parsing, and to add a flag to the vap structure (which should really be a structure/class rather than a plain vector<string>) to indicate if it's disabled.

However, all this is not needed. Upstream hostapd supports start_disabled as well. So we're not going to comment anything out.